### PR TITLE
add anomaly detection

### PR DESCRIPTION
**Pull request by double-y**
### Description
Add anomaly detection

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [x] Pass devtools::check()
- [x] Pass devtools::test()
- [x] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/anomaly_detection.R [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#' @param direction Direction of anomaly. Positive ("posi"), Negative ("neg") or "both".
+#' @param e_value Whther expected values should be returned.
+#' @param ... extra values to be passed to AnomalyDetection::AnomalyDetectionTs.
+#' @export
+do_anomaly_detection_ <- function(df, time_col, value_col, direction="both", e_value=TRUE, ...){
+  loadNamespace("dplyr")
+  loadNamespace("AnomalyDetection")
+
+  if(!direction %in% c("both", "pos", "neg")){
+    stop("derection must be 'both', 'pos' or 'neg'")
```
Comment (hideaki) [inline R/anomaly_detection.R]: Typo? : derection

File: R/anomaly_detection.R [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  value_col <- col_name(substitute(value))
+  do_anomaly_detection_(df, time_col, value_col, ...)
+}
+
+#' Detect anomaly data
+#' @param df Data frame
+#' @param time_col Column that has time data
+#' @param value_col Column that has value data
+#' @param direction Direction of anomaly. Positive ("posi"), Negative ("neg") or "both".
+#' @param e_value Whther expected values should be returned.
```
Comment (hideaki) [inline R/anomaly_detection.R]: typo: whther

File: R/anomaly_detection.R [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    # a data frame whose first column is time column
+    # and second column is value column, so it's created
+    data <- df[, c(time_col, value_col)]
+    # time column should be posixct, otherwise AnomalyDetection::AnomalyDetectionTs throws an error
+    data[[time_col]] <- as.POSIXct(data[[time_col]])
+
+    # this will be overwritten by expected values
+    expected_values <- df[[value_col]]
+
+    if(direction == "both" || direction == "pos"){
```
Comment (hideaki) [inline R/anomaly_detection.R]: positive case and negative case seems to be almost same code.
Would it be possible to share same code between them?

File: R/anomaly_detection.R (hunk 1 above)
Comment (double-y) [inline R/anomaly_detection.R]: Right, fixed it

File: R/anomaly_detection.R (hunk 2 above)
Comment (double-y) [inline R/anomaly_detection.R]: Fixed it

File: R/anomaly_detection.R (hunk 3 above)
Comment (double-y) [inline R/anomaly_detection.R]: Created an integrated function

**Pull request merged**